digest/verifiers.go

```diff
@@ -21,26 +22,26 @@ type Verifier interface {
 	// the digest.
 	Verified() bool
 
-	// Planned methods:
+	// TODO: Planned methods:
 	// Err() error
 	// Reset()
 }
 
 // NewDigestVerifier returns a verifier that compares the written bytes
 // against a passed in digest.
-func NewDigestVerifier(d Digest) Verifier {
+func NewDigestVerifier(d Digest) (Verifier, error) {
 	alg := d.Algorithm()
 	switch alg {
 	case "sha256", "sha384", "sha512":
 		return hashVerifier{
 			hash:   newHash(alg),
 			digest: d,
-		}
+		}, nil
 	default:
 		// Assume we have a tarsum.
 		version, err := tarsum.GetVersionFromTarsum(string(d))
 		if err != nil {
-			panic(err) // Always assume valid tarsum at this point.
```

Comment:
It is a programmer error to use digest.Digest(value) instead of Parse? I agree but that is a very weak enforcement and a panic is a very strong punishment.